Source/WebCore/html/parser/HTMLParserIdioms.cpp

```diff
@@ -92,10 +92,15 @@ double parseToDoubleForNumberType(StringView string, double fallbackValue)
     if (string.isEmpty())
         return fallbackValue;
 
-    // String::toDouble() accepts leading + and whitespace characters, which are not valid here.
-    UChar firstCharacter = string[0];
-    if (firstCharacter != '-' && firstCharacter != '.' && !isASCIIDigit(firstCharacter))
-        return fallbackValue;
+    // String::toDouble() accepts a set of leading whitespace characters including
+    // U+000B 'LINE TABULATION' ('\v'), but the HTML spec requirements at
+    // https://html.spec.whatwg.org/#rules-for-parsing-floating-point-number-values
+    // allow only “ASCII whitespace” — which doesn’t include U+000B, per
+    // https://infra.spec.whatwg.org/#ascii-whitespace.
+    for (uint i = 0; i < string.length(); i++) {
+        if (string[i] == '\v')
+            return fallbackValue;
+    }
```

Comment:
IIUC, that check would also incorrectly detect a `\v` in the "trailing junk" (e.g. `"5%\v"` or `"7\v"`) and reject the value altogether instead of parsing the number out of it (here, `5` or `7` respectively)

Reply:
> IIUC, that check would also incorrectly detect a `\v` in the "trailing junk" (e.g. `"5%\v"` or `"7\v"`) and reject the value altogether instead of parsing the number out of it (here, `5` or `7` respectively)

Indeed. From trying this, it’s clear that attempting to special-case `\v` at this point in the code not ideal anyway (even when updating the existing patch here to correctly handle the trailing-junk case), and that properly making WebKit conform to the spec on this will require changing WebKit’s `String::toDouble()` code (and maybe making changes to at least some of the call sites for it)